ament_cmake_python/cmake/ament_python_install_package.cmake

```diff
@@ -156,27 +160,12 @@ setup(
     DESTINATION "${ARG_DESTINATION}/${egg_install_name}.egg-info"
   )
 
-  if(ARG_SCRIPTS_DESTINATION)
-    file(MAKE_DIRECTORY "${build_dir}/scripts")  # setup.py may or may not create it
-
-    add_custom_target(
-      ament_cmake_python_build_${package_name}_scripts ALL
-      COMMAND ${python_interpreter} setup.py install_scripts -d scripts
-      WORKING_DIRECTORY "${build_dir}"
-      DEPENDS ${egg_dependencies}
-    )
-
-    if(NOT AMENT_CMAKE_SYMLINK_INSTALL)
-      # Not needed for nor supported by symlink installs
-      set(_extra_install_args USE_SOURCE_PERMISSIONS)
-    endif()
-
-    install(
-      DIRECTORY "${build_dir}/scripts/"
-      DESTINATION "${ARG_SCRIPTS_DESTINATION}/"
-      ${_extra_install_args}
-    )
-  endif()
+  # generate/install entry-point console scripts
+  foreach(_dest ${ARG_SCRIPTS_DESTINATION})
+    get_filename_component(ABS_SCRIPTS_DESTINATION "${_dest}" ABSOLUTE BASE_DIR "${CMAKE_INSTALL_PREFIX}")
+    install(CODE "execute_process(COMMAND ${python_interpreter} setup.py install_scripts --install-dir \"${ABS_SCRIPTS_DESTINATION}\"
```

Comment:
`${python_interpreter}` comes from 

https://github.com/ament/ament_cmake/blob/2bf27ce5ad01af340e1f4efd44232c1067d0dbda/ament_cmake_python/cmake/ament_python_install_package.cmake#L139

The `BUILD` argument will result in the path being a generator expression: https://github.com/ament/ament_cmake/blob/2bf27ce5ad01af340e1f4efd44232c1067d0dbda/ament_cmake_core/cmake/core/get_executable_path.cmake#L61

A workaround to make the `install(CODE` warnings go away is to call `get_executable_path` again with the `CONFIGURE` arugment.

```CMake
 get_executable_path(python_interpreter_config Python3::Interpreter CONFIGURE)
 # ...
install(CODE "execute_process(COMMAND ${python_interpreter} setup.py ...

```

The same workaround is already used a little lower in the file

https://github.com/ament/ament_cmake/blob/2bf27ce5ad01af340e1f4efd44232c1067d0dbda/ament_cmake_python/cmake/ament_python_install_package.cmake#L189-L197

Reply:
Thanks a lot for this valuable hint. I adapted the code accordingly - sorry for the long delay.
Should be ready for merging now.